javalib/src/main/scala/java/io/File.scala

```diff
@@ -323,8 +327,34 @@ class File(_path: String) extends Serializable with Comparable[File] {
 
   @stub
   def toURL(): java.net.URL = ???
-  @stub
-  def toURI(): java.net.URI = ???
+
+  // Ported from Apache Harmony
+  def toURI(): URI = {
+    import java.net.URISyntaxException
+    val path = getAbsolutePath()
+    try {
+      if (!path.startsWith("/")) {
+        // start with sep.
+        new URI(
+          "file",
+          null,
+          new StringBuilder(path.length + 1).append('/').append(path).toString,
+          null,
+          null)
+      } else if (path.startsWith("//")) {
+        // UNC path
+        new URI("file", "", path, null)
+      } else {
+        new URI("file", null, path, null, null)
+      }
+
+    } catch {
+      case e: URISyntaxException =>
+        // this should never happen
+        return null
+    }
```

Comment:
If this should never happen why is it in the code?

Reply:
Blind porting from Harmony - removed.